server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamFailureStoreDefinition.java

```diff
@@ -131,4 +134,44 @@ public class DataStreamFailureStoreDefinition {
             throw new AssertionError(e);
         }
     }
+
+    public static final String FAILURE_STORE_REFRESH_INTERVAL_SETTING_NAME = "data_streams.failure_store.refresh_interval";
+
+    public static TimeValue getFailureStoreRefreshInterval(Settings settings) {
+        return settings.getAsTime(FAILURE_STORE_REFRESH_INTERVAL_SETTING_NAME, null);
+    }
+
+    /**
+     * Like {@link DataStreamFailureStoreDefinition#applyFailureStoreSettings} but optionally applied on an existing {@link Settings}
+     * @param existingSettings initial settings to update
+     * @param nodeSettings settings from the cluster service which capture the node's current settings
+     * @return either the existing settings if no changes are needed, or a new settings instance which includes failure store specific
+     * settings
+     */
+    public static Settings buildFailureStoreIndexSettings(Settings existingSettings, Settings nodeSettings) {
+        // Optionally set a custom refresh interval for the failure store index.
+        TimeValue refreshInterval = getFailureStoreRefreshInterval(nodeSettings);
+        if (refreshInterval != null) {
+            return Settings.builder()
+                .put(existingSettings)
+                .put(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), refreshInterval)
+                .build();
+        }
+        return existingSettings;
+    }
+
+    /**
+     * Like {@link DataStreamFailureStoreDefinition#buildFailureStoreIndexSettings} but for usage with a {@link Settings.Builder}
+     * @param nodeSettings settings from the cluster service which capture the node's current settings
+     * @param builder to capture failure store specific index settings
+     * @return the original settings builder, with any failure store specific settings applied
+     */
+    public static Settings.Builder applyFailureStoreSettings(Settings nodeSettings, Settings.Builder builder) {
+        // Optionally set a custom refresh interval for the failure store index.
+        TimeValue refreshInterval = getFailureStoreRefreshInterval(nodeSettings);
+        if (refreshInterval != null) {
+            builder.put(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), refreshInterval);
```

Comment:
Nit: If this grows more, we might want to extract this on a private method so both methods can use the same "core". For now, it's so simple that I do not find a big deal if we duplicate this.

Reply:
+1 on extracting something here. I'm trying to think of a good way to do this, but I'm not loving the solutions I've come up with so far. Do we know if we're _always_ going to add custom settings to failure store indices in the (near) future? If so, that would make one of these methods redundant, which would help in refactoring 😄.

Reply:
> that would make one of these methods redundant, which would help in refactoring

I think one of these methods is currently unused, but it is used in the modify PR that will follow this. I wanted to avoid setting up

> +1 on extracting something here. I'm trying to think of a good way to do this

Very much the same for me: I was able to get it down to having zero duplication using a bunch of functional language features but I think it's far less readable like that and not significantly more compact. If it grows in the future maybe it would make sense, but for the time being I just wanted all the failure store specific configuration bits in one place.